docs/current/guides/757394-use-service-containers.md

```diff
@@ -54,167 +54,290 @@ Service containers come with the following built-in features:
+```
 
 </TabItem>
 </Tabs>
 
-In practice, you are more likely to use aliases with service bindings or endpoints, which are covered in the next section.
-
-## Expose ports
+:::tip
+Services in service containers should be configured to listen on the IP address 0.0.0.0 instead of 127.0.0.1. This is because 127.0.0.1 is only reachable within the container itself, so other services (including the Dagger health check) won't be able to connect to it. Using 0.0.0.0 allows connections to and from any IP address, including the container's private IP address in the Dagger network.
+:::
 
-Dagger offers two methods to work with service ports:
+When a service is bound to a container, it also conveys to any outputs of that container, such as files or directories. The service will be started whenever the output is used, so you can also do things like this:
 
-<Tabs groupId="language" className="embeds">
+<Tabs groupId="language">
 <TabItem value="Go">
 
-- Use the `WithExposedPort()` method to set ports that the service container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
-- Use the `Endpoint()` method to create a string address to a service container's port. You can either specify a port or let Dagger pick the first exposed port.
+```go file=./snippets/use-services/bind-services-2/main.go
+```
 
-Here's an example:
+</TabItem>
+<TabItem value="Node.js">
 
-<Embed id="kDBfYoh2uau" />
+```typescript file=./snippets/use-services/bind-services-2/index.ts
+```
 
 </TabItem>
-<TabItem value="Node.js">
+<TabItem value="Python">
 
-- Use the `withExposedPort()` method to set ports that the service container will listen on. Dagger checks the health of each exposed port prior to running any clients that use the service, so that clients don't have to implement their own polling logic.
-- Use the `endpoint()` method to create a string address to a service container's port. You can either specify a port or let Dagger pick the first exposed port.
+```python file=./snippets/use-services/bind-services-2/main.py
+```
 
-Here's an example:
+</TabItem>
+</Tabs>
 
-<Embed id="cx-3lzMDn5i" />
+### Bind host services to containers
+
+Starting with Dagger v0.8.8, you can bind host services to client containers.
+
+This type of service binding is useful when you need services on the host to communicate with one or more containers. One use case is for testing, where you need to be able to spin up ephemeral containers to run tests.
+
+Here is an example of how to instantiate host services with Dagger.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+TODO: The Dagger pipeline calls `Host.tunnel(service).start`. By default, Dagger lets the operating system randomly choose which port to use based on the available ports on the host's side. Finally, a call to `endpoint` gets the final `addr` with whichever port is bound.
```

Comment:
@vito can you please provide an example? In any one language is fine, I will convert for the other SDKs.